spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/BasicRelationalConverter.java

```diff
@@ -193,6 +193,30 @@ public Object writeValue(@Nullable Object value, TypeInformation<?> type) {
 		return conversionService.convert(value, rawType);
 	}
 
+
+	@Override
+	public boolean canWriteValue(Object value, TypeInformation<?> type) {
```

Comment:
Let's keep this method on JDBC level. The underlying assumption is that a module using the converter is able to persist entities as values.

From how the method is used, I would even suggest to keep it private and consider `public` exposure at a later time.

Reply:
The main implementation is in the `BasicRelationalConverter` based on the conversions service.

If we don't want to expose it there we could drop it completely and rely on exception handling, possibly even caching failures.

Not sure that is a good way to do it though.
I'm also not sure I understood your comment about the underlying assumption.